AcademyChallenge/Data/API/EmojiNetwork.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Network.swift
+//  AcademyChallenge
+//
+//  Created by Keven Esmael on 30/09/2022.
+//
+
+import Foundation
+
+
+class EmojiNetwork {
+	
+	
+	static func initialize() {
+	 URLSession.shared.configuration.urlCache?.diskCapacity = 100 * 1024 * 1024
+	 print("Current disk cache capacity: \(URLSession.shared.configuration.urlCache?.diskCapacity)")
+ }
+	
+	func executeNetworkCall<ResultType: Decodable>(_ call: APIProtocol, _ resultHandler: @escaping (Result<ResultType, Error>) -> Void) {
+		let decoder = JSONDecoder()
+		var request = URLRequest(url: call.url)
+		request.httpMethod = call.method.rawValue
+		call.headers.forEach { (key: String, value: String) in
+			request.setValue(value, forHTTPHeaderField: key)
+		}
+
+		let task = URLSession.shared.dataTask(with: request) { data, response, error in
+			if let data = data {
+				if let result = try? decoder.decode(ResultType.self, from: data) {
+					resultHandler(Result<ResultType, Error>.success(result))
+				} else {
+					resultHandler(Result<ResultType, Error>.failure(APIError.unknownError))
+				}
+			} else if let error = error {
+				resultHandler(Result<ResultType, Error>.failure(error))
+			}
+		}
+
+		task.resume()
+	}
+}
+
+
+
+enum Method: String {
```

Comment:
`Method`, `APIProtocol`, and `APIError` are not part of the `EmojiNetwork`, those should be moved to their own files.

Reply:
solved